system/Helpers/filesystem_helper.php

```diff
@@ -448,6 +503,24 @@ function octal_permissions(int $perms): string
 
 // ------------------------------------------------------------------------
 
+if (! function_exists('same_file'))
+{
+	/**
+	 * Checks if two files both exist and have identical hashes
+	 *
+	 * @param string $file1
+	 * @param string $file2
+	 *
+	 * @return bool  Same or not
+	 */
+	function same_file(string $file1, string $file2): bool
+	{
+		return is_file($file1) && is_file($file2) && md5_file($file1) === md5_file($file2);
```

Comment:
Use `hash_equals`

Reply:
That seems like an unnecessary measure since there are no cryptography concerns here and we already know the string lengths will match. How does this improve on `===`?